File: JAVA/src/brimore/Brimore.java
package brimore;

import java.util.Arrays;
import java.util.Scanner;

public class Brimore {
    public static void main(String[]args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] numbers = new int[N];
        for(int i = 0 ; i < N ; i++)
            numbers[i] = sc.nextInt();
        sc.close();
        Arrays.sort(numbers);
        int lowest = numbers[0];
        for(int i = 0 ; i < N ; i++)
            numbers[i] -= lowest;
        int indexLeft = 0;
        int indexRight = N-1;
        long result = numbers[N-1]-numbers[0];
        System.out.print(result + " ");
        long sumLeft = numbers[0];
        long sumsRight = numbers[N-1];
        boolean lefTurn = true;
        while(indexLeft<indexRight-1){
            if(lefTurn){
                result+= sumsRight - sumLeft;
                sumLeft+=numbers[++indexLeft];
                lefTurn= false;
            }else{
                result += sumsRight - sumLeft + numbers[indexRight-1];
                sumsRight += numbers[--indexRight];
                lefTurn = true;
            }
            System.out.print(result + " ");
        }
    }
}

File: C++/brimore_2021/Brimore.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;
int main(){
    int N;
    cin >> N;
    int arr[N];
    for(int i = 0 ; i < N ; i++)
        cin >> arr[i];
    sort(arr, arr + N);
    int indexLeft = 0;
    int indexRight = N-1;
    long long result = arr[N-1]-arr[0];
    cout << result << " ";
    long long sumLeft = arr[0];
    long long sumsRight = arr[N-1];
    bool lefTurn = true;
    while(indexLeft<indexRight-1){
        if(lefTurn){
            result+= sumsRight - sumLeft;
            sumLeft+=arr[++indexLeft];
            lefTurn= false;
        }else{
            result += sumsRight - sumLeft + arr[indexRight-1];
            sumsRight += arr[--indexRight];
            lefTurn = true;
        }
        cout << result << " ";
    }
    return 0;
}
